### PR TITLE
update dom to use newer methods

### LINKED ISSUE #113517: Use newer less verbose DOM functions
Some functions can be updated to be a bit simpler in `vs/base/browser/dom.ts`.  As far as I can tell from the linked docs and running `./scripts/test.sh` there doesn't seem to be any failures.

https://developer.mozilla.org/en-US/docs/Web/API/ChildNode/remove
`node.removeChild(node.firstChild)` -> `node.firstChild.remove()`

https://developer.mozilla.org/en-US/docs/Web/API/Node/isConnected
`isInDOM` -> `node.isConnected`

https://developer.mozilla.org/en-US/docs/Web/API/ParentNode/append
`children.forEach(child => parent.appendChild(child))` -> `parent.append(...children)`

### PR DESCRIPTION
**Pull request by nrayburn-tech**
This can be split into separate PRs if that is desired.

This PR fixes #113517

### CODE REVIEW

File: src/vs/base/browser/dom.ts [hunk 1]
```diff
@@ -1087,13 +1075,7 @@ function _$<T extends Element>(namespace: Namespace, description: string, attrs?
...
 	});
 
-	for (const child of children) {
-		if (child instanceof Node) {
-			result.appendChild(child);
-		} else if (typeof child === 'string') {
-			result.appendChild(document.createTextNode(child));
-		}
-	}
+	result.append(...children);
```
Comment (jrieken) [inline src/vs/base/browser/dom.ts:1078]: Will this implicitly create text nodes for strings?

**Comment by jrieken** [comment]
Adding @alexdima esp for the `isInDom` changes

File: src/vs/base/browser/dom.ts (hunk 1 above)
Comment (nrayburn-tech) [reply src/vs/base/browser/dom.ts:1078]: Yes.

**Review by alexdima — APPROVED** [review]
:+1: LGTM, I've also tested some standalone editor scenarios (in shadow dom, in iframe, in detached iframe, etc.)

Thank you!

**Comment by jrieken** [comment]
Thanks @nrayburn-tech

**Pull request merged**